module/modmanager/manager.go

```diff
@@ -403,6 +404,78 @@ func (mgr *Manager) ValidateConfig(ctx context.Context, conf resource.Config) ([
 	return resp.Dependencies, nil
 }
 
+// AddImplicitDependencies modifies the config diff to add implicit dependencies to changed components
+// and updates modular components & services whose module has been changed with new implict deps and adds them to conf.Modified.
+func (mgr *Manager) AddImplicitDependencies(ctx context.Context, conf *config.Diff) error {
```

Comment:
Not a huge fan of this name. Maybe something like `UpdateImplicitDependenciesInConfig` ?